app/public/js/playlists/playlistsCtrl.js

```diff
@@ -129,4 +130,32 @@ app.controller('PlaylistsCtrl', function (
             return newSize;
         }
     };
+
+    /**
+     * Responsible to shuffle a playlist
+     * instead of having to play then shuffle
+     * @param tracks [ tracks in playlist ]
+     * @method shufflePlaylist
+     */
+    $scope.shufflePlaylist = function (tracks){
```

Comment:
The intent of the function is to queue up everything in the playlist, then shuffle it, then play next in queue, more than just shuffling is necessary, I didn't come across a way of just playing an entire playlist without selecting a song to start with.